File: frontend/src/types/project.ts
export type Project = {
  id: number
  name: string
  project_type?: string | null
  owner_unit?: string | null
  supervision_unit?: string | null
  construction_unit?: string | null
  start_date?: string | null
  planned_finish_date?: string | null
  template_id?: number | null
  default_calculation_profile_id?: number | null
  default_calculation_method?: string | null
  default_baseline_plan_id?: number | null
  dashboard_config?: string | null
  report_config?: string | null
  ai_config?: string | null
  remark?: string | null
  is_archived: boolean
  archived_at?: string | null
  archive_remark?: string | null
  created_by?: string | null
  updated_by?: string | null
  created_at: string
  updated_at: string
}

export type ProjectPayload = Omit<Project, 'id' | 'created_at' | 'updated_at' | 'is_archived' | 'archived_at' | 'archive_remark'>

File: backend/app/schemas/project.py
from datetime import date, datetime

from pydantic import BaseModel, ConfigDict


class ProjectBase(BaseModel):
    name: str
    project_type: str | None = None
    owner_unit: str | None = None
    supervision_unit: str | None = None
    construction_unit: str | None = None
    start_date: date | None = None
    planned_finish_date: date | None = None
    template_id: int | None = None
    default_calculation_profile_id: int | None = None
    default_calculation_method: str | None = "auto"
    default_baseline_plan_id: int | None = None
    dashboard_config: str | None = None
    report_config: str | None = None
    ai_config: str | None = None
    remark: str | None = None
    is_archived: bool = False
    archive_remark: str | None = None
    created_by: str | None = None
    updated_by: str | None = None


class ProjectCreate(ProjectBase):
    pass


class ProjectUpdate(BaseModel):
    name: str | None = None
    project_type: str | None = None
    owner_unit: str | None = None
    supervision_unit: str | None = None
    construction_unit: str | None = None
    start_date: date | None = None
    planned_finish_date: date | None = None
    template_id: int | None = None
    default_calculation_profile_id: int | None = None
    default_calculation_method: str | None = None
    default_baseline_plan_id: int | None = None
    dashboard_config: str | None = None
    report_config: str | None = None
    ai_config: str | None = None
    remark: str | None = None
    is_archived: bool | None = None
    archive_remark: str | None = None
    updated_by: str | None = None


class ProjectArchiveRequest(BaseModel):
    archive_remark: str | None = None


class ProjectRead(ProjectBase):
    id: int
    archived_at: datetime | None = None
    created_at: datetime
    updated_at: datetime

    model_config = ConfigDict(from_attributes=True)
